tasks/sep/metal-001-MCP-L1/SignFromJson.s.sol

```diff
@@ -0,0 +1,488 @@
+    }
+
+    /// @notice Asserts that the SuperchainConfig is setup correctly
+    function checkSuperchainConfig() internal view {
+        console.log("Running assertions on the SuperchainConfig");
+
+        require(proxies.SuperchainConfig.code.length != 0, "7100");
+        require(EIP1967Helper.getImplementation(proxies.SuperchainConfig).code.length != 0, "7101");
+
+        SuperchainConfig superchainConfigToCheck = SuperchainConfig(proxies.SuperchainConfig);
+        require(superchainConfigToCheck.guardian() == superchainConfigGuardian, "7200");
+        require(superchainConfigToCheck.guardian().code.length != 0, "7250");
+        require(superchainConfigToCheck.paused() == false, "7300");
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory /* simPayload */ )
+        internal
+        view
+        override
+    {
+        console.log("Running post-deploy assertions");
+
+        checkStateDiff(accesses);
+        checkSemvers();
+
+        checkSystemConfig();
+        checkL1CrossDomainMessenger();
+        checkL1StandardBridge();
+        checkL2OutputOracle();
+        checkOptimismMintableERC20Factory();
+        checkL1ERC721Bridge();
+        checkOptimismPortal();
+        checkProtocolVersions();
+        checkSuperchainConfig();
+
+        console.log("All assertions passed!");
+    }
+
+    // This method is not storage-layout-aware and therefore is not perfect. It may return erroneous
+    // results for cases like packed slots, and silently show that things are okay when they are not.
+    function isLikelyAddressThatShouldHaveCode(uint256 value) internal pure returns (bool) {
+        // If out of range (fairly arbitrary lower bound), return false.
+        if (value > type(uint160).max) return false;
+        if (value < uint256(uint160(0x00000000fFFFffffffFfFfFFffFfFffFFFfFffff))) return false;
+
+        // If the value is a L2 predeploy address it won't have code on this chain, so return false.
+        if (
+            value >= uint256(uint160(0x4200000000000000000000000000000000000000))
+                && value <= uint256(uint160(0x420000000000000000000000000000000000FffF))
+        ) return false;
+
+        // Allow known EOAs.
+        if (address(uint160(value)) == l2OutputOracleProposer) return false;
+        if (address(uint160(value)) == l2OutputOracleChallenger) return false;
+        if (address(uint160(value)) == systemConfigOwner) return false;
+        if (address(uint160(value)) == batchSenderAddress) return false;
+        if (address(uint160(value)) == p2pSequencerAddress) return false;
+        if (address(uint160(value)) == batchInboxAddress) return false;
+
+        // Otherwise, this value looks like an address that we'd expect to have code.
+        return true;
+    }
+
+    function checkStateDiff(Vm.AccountAccess[] memory accesses) internal view {
```

Comment:
I want to figure out how to autogen `validation.md` from this function in the future. Perhaps `forge doc` + internal functions for each contract

Reply:
Another idea is that once tenderly fixes their state diff decoding (which they are working on, see more convo in slack [here](https://oplabs-pbc.slack.com/archives/C053NFWLG9K/p1711719503660909?thread_ts=1710771049.267939&cid=C053NFWLG9K)), we get the forge state diff AND the decoded tenderly state diff. Then assert the two match as a sanity check, and use the decoded tenderly one to generate the validations file